homeassistant/components/elkm1/__init__.py

```diff
@@ -175,14 +215,15 @@ def __init__(self, element, elk, elk_data):
         """Initialize the base of all Elk devices."""
         self._elk = elk
         self._element = element
+        self._prefix = elk_data['prefix']
         self._temperature_unit = elk_data['config']['temperature_unit']
         self._unique_id = 'elkm1_{}'.format(
-            self._element.default_name('_').lower())
+            self._prefix + self._element.default_name('_').lower())
```

Comment:
We can't include a configurable part in the unique id, where the user can select any variable. The unique_id have to be both unique and stable. Is there no device id we can use instead?

Reply:
Good point.  How do you feel about the CONF_HOST... it's a "permanent"ish part of the identification of the Elk M1... i.e., it's stable from the perspective of configuring home assistant -- you would have to go outside of hass to change it.  I looked through the elk M1 library and I don't see anything exposing a serial number or a mac address or anything more stable.

Reply:
If the host is an ip adress, it can change depending on router configuration, so that's not optimal good either.

Reply:
How about using the mac address instead?  We'd be taking another dependency to do that, but it's even more stable than IP address.

Beyond that possibility, I see that there is a serial number that shows up in the manufacturer's tool to program the M1, but I don't see any mention of it in the ASCII protocol documentation.  It appears it might be used as a weak form of security, where for some programming of the device you need to present the serial number to the device to allow the programming, but you cannot query the device for the programming.  @gwww any ideas?

If it is meant to be secure, then adding it to the unique id or in the config file isn't ideal.

Reply:
What about first Elk is 1. Second Elk is 2....

Reply:
Maybe you missed my edit. It is a config entry e.g.:

elk_number: <some unique id>

Where you can put in MAC, an integer (e.g.: 1), whatever (e.g: "home")

I don't know of any other identifiers that the Elk actually provides that meet the criteria of stable and unique. MAC is not something I believe you can get from the Elk.

Reply:
Using getmac module and a mac address when possible, else just using None.